frontend/src/app/core/services/authentication.service.ts

```diff
@@ -1,50 +1,103 @@
 import { Injectable } from '@angular/core';
-import {HttpService} from './http.service';
-import { UserService } from './user.service';
-import { User } from '../../shared/models/user';
-import { of } from 'rxjs';
-import { map } from 'rxjs/operators';
+import { AngularFireAuth } from '@angular/fire/auth';
 import { Router } from '@angular/router';
+import { User } from '@shared/models/user';
+import * as firebase from 'firebase';
+import { UserService } from './user.service';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
-  private user: User = null;
 
-  constructor(private router: Router, private userService: UserService) {}
+  user: firebase.User;
+  currentUser = {} as User;
+  constructor(private angularAuth: AngularFireAuth, private userService: UserService, private router: Router) {
+    this.angularAuth.authState.subscribe(user => {
+      this.configureAuthState(user);
+    });
+  }
 
-  getUser(): User {
-    if (this.user) {
-      return this.user;
+  configureAuthState(user: firebase.User): void {
+    if (user) {
+      user.getIdToken().then((theToken) => {
+        this.user = user;
+        localStorage.setItem('user', JSON.stringify(this.user));
+        localStorage.setItem('jwt', theToken);
+      });
+    }
+    else {
+      this.user = null;
     }
-    return JSON.parse(localStorage.getItem(`user`));
   }
 
-  public setUser(user: User) {
-    localStorage.setItem(`user`, JSON.stringify(user));
+  doGoogleSignIn(): Promise<void> {
+    const googleProvider = new firebase.auth.GoogleAuthProvider();
+    return this.angularAuth.signInWithPopup(googleProvider).then((auth) => {
+      this.doGoogleSignUp(auth);
+      this.router.navigate(['/portal']);
+    });
   }
 
-  public login() {
-    return this.userService.getCurrentUser().subscribe((response) => {
-      this.user = response;
-      this.setUser(this.user);
+  doGithubSignIn(): Promise<void> {
+    const githubProvider = new firebase.auth.GithubAuthProvider();
+    return this.angularAuth.signInWithPopup(githubProvider).then((auth) => {
+      this.doGithubSignUp(auth);
       this.router.navigate(['/portal']);
     });
   }
 
-  public logout() {
-    this.user = undefined;
-    this.removeUserFromStorage();
+  doGoogleSignUp(credential: firebase.auth.UserCredential) {
+    const user = {} as User;
+    user.email = credential.user.email;
+    user.username = credential.user.displayName;
+    user.avatarUrl = credential.user.photoURL;
+    user.firstName = credential.additionalUserInfo.profile[`given_name`];
+    user.lastName = credential.additionalUserInfo.profile[`family_name`];
```

Comment:
Better to write in the next style:
const user = {
    email: credential.user.email,
    username: credential.user.displayName,
    avatarUrl: credential.user.photoURL,
    firstName: credential.additionalUserInfo.profile[`given_name`],
    lastName: credential.additionalUserInfo.profile[`family_name`],
} as User;

Reply:
Do for github the same